feat: add cluster info display and reconnect capability

Ignoramuss opened the pull request:
## Summary

This PR adds cluster connection status handling and enhanced UI features to k10s:

- **Graceful disconnected state handling**: k10s no longer crashes when cluster is unavailable
- **Cluster info display**: Shows context, cluster name, namespace, and K8s version in header
- **Reconnect capability**: New `:reconnect` command to restore connection after network issues
- **Namespace switching**: Press `<0>` for all namespaces, `<d>` for default namespace
- **Enhanced TUI**: k9s-style compact header layout with keyboard shortcuts
- **Better error messaging**: Clear feedback for connection issues

## Changes

### Core Features
- Add `IsConnected()`, `Reconnect()`, and `GetClusterInfo()` methods to k8s client
- Implement connection state tracking with automatic detection of disconnects
- Add cluster info display (context, cluster, namespace, K8s version, server)
- Support starting k10s in disconnected mode (useful for demos/testing)

### UI Improvements  
- New compact header showing cluster info and keyboard shortcuts
- Namespace switching with `<0>` (all) and `<d>` (default)
- Clear visual indicators for connection status
- Integrated error/warning messages in the UI

### Developer Experience
- Updated `.gitignore` to exclude `CLAUDE.md` and `k10s-play.yaml`
- Better error handling throughout the codebase

## Testing

Tested scenarios:
- ✅ Starting k10s with valid cluster connection
- ✅ Starting k10s without cluster access (disconnected mode)
- ✅ Network interruption during operation
- ✅ Reconnecting after network restoration
- ✅ Namespace switching between all/default
- ✅ All navigation keys still work correctly

## Screenshots/Demo

The new header displays:
```
Context: minikube              <0>  All NS         <j/k>  Navigate      /\_/\
Cluster: minikube              <d>  default        <h/l>  Page        ( o.o )
K10s Rev: dev                  <:>  Command        <g/G>  Top/Bottom   > Y <
K8s Rev:  v1.28.3              <q>  Quit
CPU: n/a
MEM: n/a
```

## Related Issues

Improves usability for users working with:
- Intermittent cluster connectivity
- Multiple cluster contexts
- Development environments with frequent restarts

shvbsle commented:
I think the linter is failing. Could you fix the linter and send a revision?

shvbsle commented on internal/tui/model.go:
```diff
@@ -239,76 +323,510 @@ func (m *Model) updateTableData() {
+
+	plainText = "<g/G>  Top/Bottom"
+	styledText = shortcutStyle.Render("<g/G>") + "  " + descStyle.Render("Top/Bottom")
+	cmd2Lines = append(cmd2Lines, columnLine{styledText, plainText})
```
Have you evaluated the `Help` component provided by the Bubbles library to render this?
https://github.com/charmbracelet/bubbles?tab=readme-ov-file#help

Ignoramuss replied:
Oh ty, I was vibe coding, but this looks like exactly what I need

Ignoramuss commented on internal/tui/model.go:
```diff
@@ -171,24 +223,52 @@ func (m Model) Update(msg tea.Msg) (tea.Model, tea.Cmd) {
+					m.err = nil
+					return m, m.loadResourcesCmd(m.resourceType)
+				}
+				return m, nil
```
Ah nice catch ty

Ignoramuss commented:
> I think the linter is failing. Could you fix the linter and send a revision?

Yes!

Ignoramuss commented:
Hey! Just pushed a revision with all the feedback addressed.

**Linter is fixed** - turned out there were missing godoc comments on exported symbols. Added them across all packages (k8s, tui, config, main). Also fixed a couple other linter complaints: file.Close() wasn't being checked in config.go, and there was an unused max() function in model.go.

**Re: logging location** - Logs go to ~/.local/state/k10s/k10s.log using the XDG state directory. Chose this over /var/log since it's cross-platform and doesn't need root. The adrg/xdg library handles the platform-specific paths automatically.

**Re: Help component** - Good catch, switched to using the bubbles help component instead of manually rendering shortcuts. Much cleaner.

**Re: namespace switch panic** - Fixed by resetting the paginator page to 0 when switching namespaces. Without it, if you're on page 2 of pods in namespace A and switch to namespace B which only has 1 page, you get a slice bounds panic.

Also improved the Makefile while I was at it - added a `check` target that runs fmt/vet/lint, and made `build` depend on it. Should catch issues earlier now.

Linter passes clean locally:
```
$ make build
go fmt ./...
go vet ./...
golangci-lint run
0 issues.
go build ... 
```

Ignoramuss commented on internal/k8s/client.go:
```diff
@@ -38,12 +48,115 @@ func NewClient() (*Client, error) {
+	if c.config == nil {
+		config, err := getKubeConfig()
+		if err != nil {
+			return fmt.Errorf("failed to get kubeconfig: %w", err)
```
These would be printed in the calling method through the logger

shvbsle reviewed (approved):
LGTM
